infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -159,8 +161,8 @@ public CorfuLogReplicationRuntime(LogReplicationRuntimeParameters parameters, Lo
         this.sourceManager = new LogReplicationSourceManager(parameters, new LogReplicationClient(router, remoteClusterId),
             metadataManager);
         this.connectedEndpoints = new HashSet<>();
-        this.communicationFSMWorkers = Executors.newSingleThreadExecutor(new
-                ThreadFactoryBuilder().setNameFormat("runtime-fsm-worker").build());
+        this.communicationFSMWorkers = new ThreadPoolExecutor(1, 1, 0L,
+                TimeUnit.MILLISECONDS, new LinkedBlockingQueue<Runnable>());
```

Comment:
Do we need to set thread name by a threadFactory?

Reply:
Yes, I think this is a good idea, so we track it easier. Added.